File: prototype/Assets/microcosmicWar/Scripts/EmitterContainer.js


///var fireSound:AudioSource;


class EmitterContainer  extends Emitter
{
	var emitList:Emitter[];

	virtual function setInjureInfo(pInjureInfo:Hashtable)
	{
		super.setInjureInfo( pInjureInfo );
		for( var iEmit:Emitter in emitList )
		{
			iEmit.setInjureInfo(pInjureInfo);
		}
	}

	function Start()
	{
	}

	virtual function EmitBullet()
	{
		for( var iEmit:Emitter in emitList )
		{
			iEmit.EmitBullet();
		}
		if(fireSound)
		{
			fireSound.Play();
		}
	}

	virtual function setBulletLayer(pBulletLayer:int)
	{
		//print("EmitterContainer.setBulletLayer");
		for( var iEmit:Emitter in emitList )
		{
			iEmit.setBulletLayer(pBulletLayer);
		}
	}
}

File: prototype/Assets/microcosmicWar/Scripts/EmitterContainer.cs

using UnityEngine;
using System.Collections;
using System.Collections.Generic;


///AudioSource fireSound;


public class EmitterContainer : Emitter
{
    public Emitter[] emitList;

    public override void setInjureInfo(Hashtable pInjureInfo)
    {
        base.setInjureInfo(pInjureInfo);
        foreach (Emitter iEmit in emitList)
        {
            iEmit.setInjureInfo(pInjureInfo);
        }
    }

    //void  Start (){
    //}

    public override GameObject[] EmitBullet()
    {
        List<GameObject> lOut = new List<GameObject>();
        foreach (Emitter iEmit in emitList)
        {
            lOut.AddRange(iEmit.EmitBullet());
        }
        if (fireSound)
        {
            fireSound.Play();
        }
        playFireSpark();
        return lOut.ToArray();
    }

    public override void setBulletLayer(int pBulletLayer)
    {
        //print("EmitterContainer.setBulletLayer");
        foreach (Emitter iEmit in emitList)
        {
            iEmit.setBulletLayer(pBulletLayer);
        }
    }
}
